web/src/helpers/formatting.js

```diff
@@ -16,38 +17,89 @@ const formatCo2 = function (d, numDigits) {
   d /= 1e6; // Convert to tCO₂ / min
   if (d == null || d === NaN) return d;
   if (numDigits == null) numDigits = 3;
-  if (d >= 1) // a ton or more
+  if (d >= 1)
+    // a ton or more
     return d3.format('.' + numDigits + 's')(d) + 't ' + translation.translate('ofCO2eqPerMinute');
-  else
-    return d3.format('.' + numDigits + 's')(d * 1e6) + 'g ' + translation.translate('ofCO2eqPerMinute');
+  else return d3.format('.' + numDigits + 's')(d * 1e6) + 'g ' + translation.translate('ofCO2eqPerMinute');
 };
 const scalePower = function (maxPower) {
   // Assume MW input
   if (maxPower < 1)
     return {
-      unit: "kW",
-      formattingFactor: 1e-3
-    }
+      unit: 'kW',
+      formattingFactor: 1e-3,
+    };
   if (maxPower < 1e3)
     return {
-      unit: "MW",
-      formattingFactor: 1
-    }
-  else return {
-      unit: "GW",
-      formattingFactor: 1e3
-    }
+      unit: 'MW',
+      formattingFactor: 1,
+    };
+  else
+    return {
+      unit: 'GW',
+      formattingFactor: 1e3,
+    };
+};
+
+const formatDate = function (date, lang, time) {
+  if (!date || !time) return '';
+
+  switch (time) {
+    case TIME.HOURLY:
+      return new Intl.DateTimeFormat(lang, { dateStyle: 'long', timeStyle: 'short' }).format(date);
+    case TIME.DAILY:
+      return new Intl.DateTimeFormat(lang, { dateStyle: 'long' }).format(date);
+    default:
+      console.error(`${time} is not implemented`);
+      return '';
+  }
 };
 
-const formatHourlyDate = function (date, lang) {
-  // formats date object to readable date
-  if (!date) return '';
-  return new Intl.DateTimeFormat(lang, {dateStyle: 'long', timeStyle: 'short' }).format(date);
+const getLocaleUnit = (dateUnit, lang) =>
```

Comment:
Nice hack :)
Were you able to test in older Safari versions?

Reply:
No I can't seem to figure out how to do that. Maybe we save that for final QA?